backend/src/packages/users/user.repository.ts

```diff
@@ -26,6 +26,17 @@ class UserRepository implements IRepository {
     return Promise.resolve([]);
   }
 
+  public async findByPhone(value: string): Promise<UserEntity | null> {
+    const result = await this.db
+      .driver()
+      .select()
+      .from(this.usersSchema)
+      .where(eq(this.usersSchema.phone, value))
+      .execute();
+
+    return result[0] ? UserEntity.initialize(result[0]) : null;
```

Comment:
Let's merge it now, but refactor later.
```suggestion
    const [user] = await this.db
      .driver()
      .select()
      .from(this.usersSchema)
      .where(eq(this.usersSchema.phone, value))
      .execute();

    return user ? UserEntity.initialize(user) : null;
```

Reply:
I had the same variant of code, but it doesn't pass linting:
  37:12  warning  Unnecessary conditional, value is always truthy  @typescript-eslint/no-unnecessary-condition
  ✖ 1 problem (0 errors, 1 warning)
ESLint found too many warnings (maximum: 0).